### ISSUE TITLE
(no pycaffe) Segmentation Fault - CPU Execution

### ISSUE THREAD

**Issue by isn4** [BVLC/caffe#5462]
Hello,
I've looked around at other issues submitted to the caffe-segnet issues page that are similar to this one and can only find advice that falls flat. I haven't seen the issue I opened there get much attention so I'm hoping I can find a solution here. At this point in my search for answers it seems as though I've come across a bug in the caffe code rather than in the model I'm using. I suspect this is the case because when I run caffe-segnet (a branch of caffe) on my data using SegNet and CPU mode I reach a segmentation fault.
If anyone is familiar with SegNet, feel free to see the issue I opened on their repository, [here](https://github.com/alexgkendall/caffe-segnet/issues/101). I've followed all of the suggested guidelines to fit my data to their net so I'm beginning to suspect the problem is not rooted there.
On this repository is where I'd like to ask for help regarding caffe itself. Is there anyone who can help me trace this segmentation fault back to its root?

### Here is the command I use to run caffe-segnet:
./build/tools/caffe train -solver tutorial/Models/segnet_solver.prototxt

### My stack trace:
I0328 16:04:58.364321 29751 net.cpp:482] Collecting Learning Rate and Weight Decay.
I0328 16:04:58.364439 29751 net.cpp:247] Network initialization done.
I0328 16:04:58.364537 29751 net.cpp:248] Memory required for data: 1064448016
I0328 16:04:58.364912 29751 solver.cpp:42] Solver scaffolding done.
I0328 16:04:58.365309 29751 solver.cpp:250] Solving VGG_ILSVRC_16_layer
I0328 16:04:58.365383 29751 solver.cpp:251] Learning Rate Policy: step
*** Aborted at 1490731504 (unix time) try "date -d @1490731504" if you are using GNU date ***
PC: @     0x2ad61541b15d caffe::SoftmaxWithLossLayer<>::Forward_cpu()
*** SIGSEGV (@0x2ad74b84c410) received by PID 29751 (TID 0x2ad61507ee40) from PID 1266992144; stack trace: ***
    @     0x2ad616439100 (unknown)
    @     0x2ad61541b15d caffe::SoftmaxWithLossLayer<>::Forward_cpu()
    @     0x2ad61543fd41 caffe::Net<>::ForwardFromTo()
    @     0x2ad615440097 caffe::Net<>::ForwardPrefilled()
    @     0x2ad61545d915 caffe::Solver<>::Step()
    @     0x2ad61545e394 caffe::Solver<>::Solve()
    @           0x409985 train()
    @           0x4073e0 main
    @     0x2ad624110b15 __libc_start_main
    @           0x407949 (unknown)
    @                0x0 (unknown)
/var/slurmd/job962498/slurm_script: line 39: 29751 Segmentation fault
caffe train -solver /usr/caffe-segnet/tutorial/Models/segnet_solver.prototxt

### My system configuration
Operating system: working on a remote machine
Compiler: gcc 4.8.4
Python version: python/2.7.11

Thanks in advance!

**Comment by williford** [comment]
Please make the problem reproducible using the main Caffe fork and then re-open this issue.

**Comment by isn4** [comment]
Resolved [here](https://github.com/alexgkendall/caffe-segnet/issues/101).
